pkg/controllers/statusaggregator/plugins/deployment.go

```diff
@@ -45,16 +50,38 @@ func (receiver *DeploymentPlugin) AggregateStatues(
 ) (*unstructured.Unstructured, bool, error) {
 	logger := klog.FromContext(ctx).WithValues("status-aggregator-plugin", "deployments")
 
-	needUpdate := false
+	needUpdate, needUpdateObservedGeneration := false, true
 	digests := []util.LatestReplicasetDigest{}
 
 	sourceDeployment := &appsv1.Deployment{}
 	if err := util.ConvertViaJson(sourceObject, sourceDeployment); err != nil {
 		return nil, false, err
 	}
 
-	aggregatedStatus := &appsv1.DeploymentStatus{
-		ObservedGeneration: sourceDeployment.Generation,
+	aggregatedStatus := &appsv1.DeploymentStatus{}
+
+	resource := &fedtypesv1a1.GenericObjectWithStatus{}
+	err := util.UnstructuredToInterface(fedObject, resource)
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to unmarshall to generic resource: %w", err)
+	}
+
+	templateSynced, err := checkResourceTemplateSyncStatus(sourceObject, fedObject, resource)
+	if err != nil {
+		logger.Error(err, "Failed to check resource template sync status")
+		return nil, false, err
+	}
+
+	if !templateSynced {
+		logger.V(3).Info("Resource template is not synced")
+		needUpdateObservedGeneration = false
+	}
+
+	collectGenerations := make(map[string]int64)
+	if resource.Status != nil {
```

Comment:
@gary-lgy If `resource.Status` is nil, does it imply that the federated object has not been synced once? If that is the case, can we check this above and terminate early?